com.microsoft.mrtk.input/Experimental/SpatialMouse/Interactor/SpatialMouseInteractor.cs

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Subsystems;
+using System.Collections.Generic;
+using Unity.Profiling;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.XR.Interaction.Toolkit;
+using UnityEngine.XR.Interaction.Toolkit.Inputs;
+using UnityEngine.XR.Interaction.Toolkit.UI;
+
+namespace Microsoft.MixedReality.Toolkit.Input.Experimental
+{
+    /// <summary>
+    /// An extension of XRRayInteractor which has extra functionality for handling spatial mouse movement and scrolling. 
+    /// </summary>
+    /// <remarks>
+    /// <remarks>
+    /// This is an experimental feature. This class is early in the cycle, it has 
+    /// been labeled as experimental to indicate that it is still evolving, and 
+    /// subject to change over time. Parts of the MRTK, such as this class, appear 
+    /// to have a lot of value even if the details haven�t fully been fleshed out. 
+    /// For these types of features, we want the community to see them and get 
+    /// value out of them early enough so to provide feedback. 
+    /// </remarks>
+    [AddComponentMenu("Scripts/Microsoft/MRTK/Input/MRTK Spatial Mouse Interactor")]
+    public class SpatialMouseInteractor : XRRayInteractor, IRayInteractor
+    {
+        [SerializeField, Experimental]
+        [Tooltip("The Input System action used for mouse movement. Must be a Vector2 Control.")]
+        private InputActionProperty mouseMoveAction;
```

Comment:
Please follow the pattern used elsewhere in MRTK3.   Create "private serializable fields" and "public properties" to expose these fields.

For example

```
        [SerializeField, Experimental]
        [Tooltip("The Input System action used for mouse movement. Must be a Vector2 Control.")]
        private InputActionProperty mouseMoveAction;

        /// <summary>
        /// The Input System action used for mouse movement. Must be a Vector2 Control.
        /// </summary>
        public InputActionProperty MouseMoveAction 
        {
             get => mouseMoveAction;
             set => mouseMoveAction = value;
        }
```

Alternatively, you can create serializable public property with the "field" attribute.

```
        /// <summary>
        /// The Input System action used for mouse movement. Must be a Vector2 Control.
        /// </summary>
        [field: SerializeField, Tooltip(" The Input System action used for mouse movement. Must be a Vector2 Control..")]
         public InputActionProperty MouseMoveAction  { get; set; }
```